advisories/a79281.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Technical Advisory 79281
+summary: Importing duplicate keys can cause violations of UNIQUE constrants
+toc: true
+docs_area: releases
+---
+
+Publication date: April 12, 2022
+
+## Description
+
+In a multi-region database, calling [`IMPORT INTO`](../v22.1/import-into.html) on a [`REGIONAL BY ROW` table](../v22.1/regional-tables.html) could violate [`UNIQUE`](../v22.1/unique.html) constraints, including [`PRIMARY KEY`](../v22.1/primary-key.html) constraints, resulting in duplicate keys.
+
+Affected versions: v21.2.0-v21.2.7
+
+## Statement
+
+This is resolved in CockroachDB by [PR 79293](https://github.com/cockroachdb/cockroach/pull/79293).
+The fix has been applied to the maintenance release of [CockroachDB v21.2.8](../releases/v21.2.html#v21-2-8).
+
+This public issue is tracked by [79281](https://github.com/cockroachdb/cockroach/issues/79281).
+
+## Mitigation
+
+Users of CockroachDB v21.2.0 to v21.2.7 who have used [`IMPORT INTO`](../v22.1/import-into.html) should audit their [`REGIONAL BY ROW`](../v22.1/regional-tables.html) tables with `UNIQUE` constraints to ensure these tables do not contain `UNIQUE` constraint violations.
+
+Users should also upgrade to v21.2.8 or a later version.
+
+### v21.2.6 or later
+
+Users of CockroachDB v21.2.6 and later have access to the following [built-in functions](../v22.1/functions-and-operators.html), which can be used to easily validate unique constraints:
+
+- `SELECT crdb_internal.revalidate_unique_constraints_in_all_tables();`
+- `SELECT crdb_internal.revalidate_unique_constraints_in_table('table_name');`
+- `SELECT crdb_internal.revalidate_unique_constraint('table_name', 'constraint_name');`
+
+If any constraint fails validation, the functions will return an error with a hint about which data caused the constraint violation. These violations can then be resolved manually by updating or deleting the rows in violation.
+
+### Prior to v21.2.6
+
+Users of CockroachDB releases prior to v21.2.6 may audit the table for duplicate keys by running a query of the following form each [`UNIQUE`](../v22.1/unique.html) constraint:
```

Comment:
I think this is meant to say "... a query of the following for each..." Maybe a typo on "form"